storage/boltdb/client.go

```diff
@@ -74,6 +74,10 @@ func (c *Client) Get(pathKey storage.Key) (storage.Value, error) {
 	err := c.db.Update(func(tx *bolt.Tx) error {
 		b := tx.Bucket(c.Bucket)
 		v := b.Get(pathKey)
+		if len(v) == 0 {
+			return Error.New("This file does not exist")
```

Comment:
I'd suggest to return a special Err variable here, e.g. `ErrKeyNotFound`.

We can define it at /pkg/storage/common.go like this:
```go
var ErrKeyNotFound = errors.New("key not found")
```
This will allow callers to handle it gracefully. This way the MinioGW's GetObjectInfo method will check for it and return `minio.ObjectNotFound{Bucket: bucket, Object: object}` so the S3 clients can report a user friendly error instead of "500 Internal Error".

Reply:
you could use a class for this, such as ErrKeyNotFound.New("file does not exist")

somewhere ErrKeyNotFound = errs.Class("key not found")

later you can test ErrKeyNotFound.Has(err)

Reply:
thanks!